From 26205f1f656c84fefaecbdd5b57feeed23e4f259 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Thu, 12 Sep 2024 11:53:11 +0300
Subject: [PATCH] [FSW-6580] fs1: init boot info after recovery

After recovery, boot info set done flag is true
but not valid for newly updated image in
recovery.
Hence, after recovery boot rom has to write new
boot info when validation flag is not set as it
was cleared during recovery.

Change-Id: I10b780e51635ecf0ac4172638e86faf2e2965948
---

diff --git a/arch/fs1/arc/dev_boot_info.c b/arch/fs1/arc/dev_boot_info.c
index de498e6..c9bfd67 100644
--- a/arch/fs1/arc/dev_boot_info.c
+++ b/arch/fs1/arc/dev_boot_info.c
@@ -36,7 +36,7 @@
 	if(boot_rom_skip_boot_info())
 		return;
 
-	if (!boot_rom_global.boot_info_set_done) {
+	if (!boot_rom_global.boot_info_set_done || !boot_rom_global.validation_pass) {
 		boot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_START);
 
 		/* TODO: save boot info required for other FW modules */
